userspace/libsinsp/threadinfo.h

```diff
@@ -65,6 +65,78 @@ struct erase_fd_params
 class SINSP_PUBLIC sinsp_threadinfo : public libsinsp::state::table_entry
 {
 public:
+	class sinsp_userinfo
+	{
+	public:
+		sinsp_userinfo() {
+			m_uid = 0xffffffff;
+			m_gid = 0xffffffff;
+		}
+		uint32_t uid() { return m_uid; }
+		uint32_t gid() { return m_gid; }
+		std::string name() {
```

Comment:
Besides all the compiler optimizations here, why can't this return `const std::string&`? This can be done by defining the const strings you have here as static class members.

Copying the string around in many places seems a bit overkill to me. Evidence is that you had to define storage strings in many different places. All those string copies will have a cost at runtime when evaluating the filter. WDYT?

Also, I'd advicate for having all the getters being `inline` and `const`.

Reply:
> why can't this return const std::string&? This can be done by defining the const strings you have here as static class members.

This will bring us back to the discussion of having to store "N/A" instead of an empty string and the corresponding overhead, which I would like to avoid.

> Copying the string around in many places seems a bit overkill to me. Evidence is that you had to define storage strings in many different places. All those string copies will have a cost at runtime when evaluating the filter.

Which costs do you mean, memory or CPU utilization? The only point where I see this being called is indeed filtering evaluation, it's already minimizes the impact to only one invocation per object. In the end I would anyway expect it to be hidden away by either copy elision or NRVO (depending on the version of C++ standard being used). I can take a look at the produced code in more details.

> I'd advicate for having all the getters being inline and const.

What for do you want to have an inline? Those getters are not going to be included multiple times in any form, and compiler can surely find out better if they have to be inlined or not. Looking through the specs, they even say that a function fully defined in the class definition would be implicitly an inline one. Adding const is a good idea, will do that.

Reply:
I've added the const qualifier.